### PR TITLE
Use new System.Net.ServerSentEvents package in SignalR

### PR DESCRIPTION
**Pull request by BrennanConroy**
Main differences:
1. `\r` is handled now, before we didn't add support for just `\r`
2. We used to use `CancelPendingRead()` to trigger cancellation, now we get an `OperationCanceledException`
3. `FormatException` is no longer thrown when a partial message is received and the HTTP request is done. I believe that is more spec compliant.
4. When a message is split across multiple data frames, a `\n` character is inserted into the final message between each data frame. Our old implementation used `Environment.NewLine` but the spec mentions `\n`

### CODE REVIEW

**Review by wtgodbe — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
Nice. Thanks.

File: src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/ServerSentEventsTransport.cs [hunk 1]
```diff
@@ -130,80 +129,35 @@ private async Task ProcessEventStream(HttpResponseMessage response, Cancellation
...
 #pragma warning restore CA2016 // Forward the 'CancellationToken' parameter to methods
         {
-            var reader = PipeReader.Create(stream);
-
-            using var registration = cancellationToken.Register(CancelReader, reader);
-
             try
             {
-                while (true)
+                var parser = SseParser.Create(stream, (eventType, bytes) => bytes.ToArray());
```
Comment (davidfowl) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/ServerSentEventsTransport.cs:139]: What's this array allocation?

**Review by davidfowl — APPROVED** [review]
Code deletion is so nice!

**Pull request merged**